cmd/compute-domain-kubelet-plugin/nvlib.go

```diff
@@ -319,65 +312,21 @@ func (l deviceLib) parseNVCapDeviceInfo(nvcapsFilePath string) (*nvcapDeviceInfo
 	return info, nil
 }
 
-func (l deviceLib) createComputeDomainChannelDevice(channel int) error {
-	// Construct the properties of the device node to create.
-	path := fmt.Sprintf("/dev/nvidia-caps-imex-channels/channel%d", channel)
-	path = filepath.Join(l.devRoot, path)
-	mode := uint32(unix.S_IFCHR | 0666)
-
-	// Get the IMEX channel major and build a /dev device from it
+func (l deviceLib) getNVCapIMEXChannelDeviceInfo(channelID int) (*nvcapDeviceInfo, error) {
 	major, err := l.getDeviceMajor(nvidiaCapsImexChannelsDeviceName)
 	if err != nil {
-		return fmt.Errorf("error getting IMEX channel major: %w", err)
-	}
-	dev := unix.Mkdev(uint32(major), uint32(channel))
-
-	// Recursively create any parent directories of the channel.
-	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
-		return fmt.Errorf("error creating directory for IMEX channel device nodes: %w", err)
-	}
-
-	// Remove the channel if it already exists.
-	if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("error removing existing IMEX channel device node: %w", err)
-	}
-
-	// Create the device node using syscall.Mknod
-	if err := unix.Mknod(path, mode, int(dev)); err != nil {
-		return fmt.Errorf("mknod of IMEX channel failed: %w", err)
-	}
-
-	return nil
-}
-
-func (l deviceLib) createNvCapDevice(nvcapFilePath string) error {
-	// Get the nvcapDeviceInfo for the nvcap file.
-	deviceInfo, err := l.parseNVCapDeviceInfo(nvcapFilePath)
-	if err != nil {
-		return fmt.Errorf("error parsing nvcap file for fabric-imex-mgmt: %w", err)
-	}
-
-	// Construct the necessary information to create the device node
-	path := filepath.Join(l.devRoot, deviceInfo.path)
-	mode := unix.S_IFCHR | uint32(deviceInfo.mode)
-	dev := unix.Mkdev(uint32(deviceInfo.major), uint32(deviceInfo.minor))
-
-	// Recursively create any parent directories of the device.
-	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
-		return fmt.Errorf("error creating directory for nvcaps device nodes: %w", err)
-	}
-
-	// Remove the device if it already exists.
-	if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("error removing existing nvcap device node: %w", err)
+		return nil, fmt.Errorf("error getting device major: %w", err)
 	}
 
-	// Create the device node using syscall.Mknod
-	if err := unix.Mknod(path, mode, int(dev)); err != nil {
-		return fmt.Errorf("mknod of nvcap device failed: %w", err)
+	info := &nvcapDeviceInfo{
+		major:  major,
+		minor:  channelID,
+		mode:   0666,
```

Comment:
Did we think hard about 666, ever? Why not 777? :) Why give privileges to 'others'? Not important questions here, I think. Just generally curious from a security / polp pov.

Reply:
connecting the dots: https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/508#discussion_r2308246292